commit d8974f05792a102b5e2e5df21427bb64aa59f726
Author: Simon Morlat <simon.morlat@linphone.org>
Date:   Wed Nov 12 18:04:50 2025 +0100

    Fix: race condition resulting in a NullPointerException on Android, when the audio focus is lost while the Core is being stopped.

diff --git a/liblinphone/wrappers/java/classes/org/linphone/core/tools/AndroidPlatformHelper.java b/liblinphone/wrappers/java/classes/org/linphone/core/tools/AndroidPlatformHelper.java
index 3ec4086ad3..9044597264 100644
--- a/liblinphone/wrappers/java/classes/org/linphone/core/tools/AndroidPlatformHelper.java
+++ b/liblinphone/wrappers/java/classes/org/linphone/core/tools/AndroidPlatformHelper.java
@@ -1519,21 +1519,32 @@ public class AndroidPlatformHelper {
         mAudioHelper.restorePreviousAudioRoute();
     }
 
+    public void onAudioFocusLostFromCoreThread(){
+        if (mCore == null) return;
+        boolean pauseCallsWhenAudioFocusIsLost = mCore.getConfig().getBool("audio", "android_pause_calls_when_audio_focus_lost", true);
+        if (pauseCallsWhenAudioFocusIsLost) {
+            if (mCore.isInConference()) {
+                Log.i("[Platform Helper] App has lost audio focus, leaving conference");
+                leaveConference(mCore.getNativePointer());
+            } else {
+                Log.i("[Platform Helper] App has lost audio focus, pausing all calls");
+                pauseAllCalls(mCore.getNativePointer());
+            }
+            mAudioHelper.releaseCallAudioFocus();
+        } else {
+            Log.w("[Platform Helper] Audio focus lost but keeping calls running");
+        }
+    }
+
     public void onAudioFocusLost() {
         if (mCore != null) {
-            boolean pauseCallsWhenAudioFocusIsLost = mCore.getConfig().getBool("audio", "android_pause_calls_when_audio_focus_lost", true);
-            if (pauseCallsWhenAudioFocusIsLost) {
-                if (mCore.isInConference()) {
-                    Log.i("[Platform Helper] App has lost audio focus, leaving conference");
-                    leaveConference(mCore.getNativePointer());
-                } else {
-                    Log.i("[Platform Helper] App has lost audio focus, pausing all calls");
-                    pauseAllCalls(mCore.getNativePointer());
+            Runnable focusLostRunnable = new Runnable(){
+                @Override
+                public void run() {
+                    onAudioFocusLostFromCoreThread();
                 }
-                mAudioHelper.releaseCallAudioFocus();
-            } else {
-                Log.w("[Platform Helper] Audio focus lost but keeping calls running");
-            }
+            };
+            dispatchOnCoreThread(focusLostRunnable);
         }
     }
 
